packages/components/src/Actions/Actions.component.js

```diff
@@ -49,32 +53,27 @@ const actions: [
  */
 function Actions(props) {
 	const buttonGroupProps = getButtonGroupProps(props);
+	const globalParams = {
+		hideLabel: props.hideLabel,
+		link: props.link,
+		tooltipPlacement: props.tooltipPlacement,
+	};
+
 	return (
 		<ButtonGroup className="tc-actions" {...buttonGroupProps}>
 			{props.actions.map((action, index) => {
 				const { displayMode, ...rest } = action;
-				switch (displayMode) {
-				case TYPE_DROPDOWN:
-					return (
-						<ActionDropdown
-							hideLabel={props.hideLabel}
-							key={index}
-							link={props.link}
-							tooltipPlacement={props.tooltipPlacement}
-							{...rest}
-						/>
-					);
-				default:
-					return (
-						<Action
-							hideLabel={props.hideLabel}
-							key={index}
-							link={props.link}
-							tooltipPlacement={props.tooltipPlacement}
-							{...rest}
-						/>
-					);
-				}
+				const params = {
+					key: index,
+					...globalParams,
+					...rest,
+				};
+				const ActionType = {
+					[TYPE_DROPDOWN]: ActionDropdown,
+					[TYPE_SPLIT_DROPDOWN]: ActionSplitDropdown,
+				}[displayMode] || Action;
+
+				return <ActionType {...params} />;
```

Comment:
The ActionType looks smaller indeed but it makes the code harder to understand and does an extra allocation for every resolution of the ActionType. 
Id rather see a function returning the ActionType Component with a proper switch/case.